examples/parsing/explicit_curry.an

```diff
@@ -0,0 +1,17 @@
+
+foo _ a
+oof 1 _ 3
+
+bar _ "test" _
+
+baz 1 (_, 32)
+qux 3 (_) _
+
+
+// args: --parse
+// expected stdout:
+// (\ _$1 . (foo _$1 a));
+// (\ _$1 . (oof 1 _$1 3));
+// (\ _$1 _$2 . (bar _$1 "test" _$2));
+// (baz 1 (',' _ 32));
```

Comment:
I'd expect `(',' _ 32)` to be a lambda here - `(\ $1 . (',' $1 32))` (see comment below)

Reply:
Also, I hate to be ante-fun but can we change oof's name to something else? Following foo, bar, baz, `qux` usually comes after those three

Reply:
I had written qux originally, but I dont remember why i changed it. Will change it back :)

Reply:
> 
> 
> I'd expect `(',' _ 32)` to be a lambda here - `(\ $1 . (',' $1 32))` (see comment below)
But the curried arg shouldn't bubble up to the sorrounding lambda, right? And should the explicit currying happen at arbitrary depths?

Reply:
Correct, it shouldn't bubble up it should only happen for the function call its used in which in this case looks to be `,`. So the second argument of `baz` should be this function

Reply:
Since `,` is a type constructor this case may just not be desugared yet because of the temporary type constructor check